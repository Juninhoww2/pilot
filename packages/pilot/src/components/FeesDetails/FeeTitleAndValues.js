import React from 'react'
import PropTypes from 'prop-types'
import { Flexbox } from 'former-kit'
import formatPercent from '../../formatters/percent'
import formatCurrency from '../../formatters/currency'
import styles from './styles.css'

const formatterByType = {
  currency: formatCurrency,
  percent: formatPercent,
}

const FeeTitleAndValues = ({ t, title, values }) => (
  <div>
    <p className={styles.title}>{title}</p>
    <Flexbox className={styles.feeValuesWrapper}>
      {values.map(({ fees, translationPath }) => (
        <div key={translationPath}>
          <p>{t(translationPath)}</p>
          <p>
            <strong>
              {
                fees
                  .map(({ type, value, valueSuffixPath }) => {
                    const formatter = formatterByType[type]
                    const formattedValue = formatter(value) || 'N/A'
                    const valueSuffix = valueSuffixPath ? t(valueSuffixPath) : ''

                    return `${formattedValue} ${valueSuffix}`
                  })
                  .join('+ ')
              }
            </strong>
          </p>
        </div>
      ))}
    </Flexbox>
  </div>
)

FeeTitleAndValues.propTypes = {
  t: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  values: PropTypes.arrayOf(PropTypes.shape({
    fees: PropTypes.arrayOf(PropTypes.shape({
      type: PropTypes.string.isRequired,
      value: PropTypes.number,
      valueSuffixPath: PropTypes.string,
    })),
    translationPath: PropTypes.string.isRequired,
  })).isRequired,
}

export default FeeTitleAndValues
