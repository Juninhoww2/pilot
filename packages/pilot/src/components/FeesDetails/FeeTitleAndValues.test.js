import React from 'react'
import { identity } from 'ramda'
import { render } from '@testing-library/react'
import FeeTitleAndValues from './FeeTitleAndValues'
import { buildPixValues } from '.'

test('render pix payment spread fee', () => {
  const fees = {
    pix: {
      paymentFixedFee: 0,
      paymentSpreadFee: 1.19,
    },
  }

  const {
    container,
    getByText,
  } = render(
    <FeeTitleAndValues
      title="pages.empty_state.fees.paid"
      values={[{
        fees: buildPixValues(fees),
        translationPath: 'pages.empty_state.fees.pix',
      }]}
      t={identity}
    />
  )

  getByText(/pages.empty_state.fees.paid/)
  getByText(/pages.empty_state.fees.pix/)
  getByText('1,19%')
  expect(container).toMatchSnapshot()
})

test('render pix payment fixed fee', () => {
  const fees = {
    pix: {
      paymentFixedFee: 380,
      paymentSpreadFee: 0,
    },
  }

  const {
    container,
    getByText,
  } = render(
    <FeeTitleAndValues
      title="pages.empty_state.fees.paid"
      values={[{
        fees: buildPixValues(fees),
        translationPath: 'pages.empty_state.fees.pix',
      }]}
      t={identity}
    />
  )

  getByText(/pages.empty_state.fees.paid/)
  getByText(/pages.empty_state.fees.pix/)
  getByText('R$ 3,80')
  expect(container).toMatchSnapshot()
})

test('render pix payment fixed fee and spread fees', () => {
  const fees = {
    pix: {
      paymentFixedFee: 380,
      paymentSpreadFee: 1.19,
    },
  }

  const {
    container,
    getByText,
  } = render(
    <FeeTitleAndValues
      title="pages.empty_state.fees.paid"
      values={[{
        fees: buildPixValues(fees),
        translationPath: 'pages.empty_state.fees.pix',
      }]}
      t={identity}
    />
  )

  getByText(/pages.empty_state.fees.paid/)
  getByText(/pages.empty_state.fees.pix/)
  expect(container).toMatchSnapshot()
})

