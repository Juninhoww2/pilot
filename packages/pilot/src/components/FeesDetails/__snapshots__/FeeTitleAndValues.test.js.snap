// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render pix payment fixed fee 1`] = `
<div>
  <div>
    <p
      class="title"
    >
      pages.empty_state.fees.paid
    </p>
    <div
      class="feeValuesWrapper"
    >
      <div>
        <p>
          pages.empty_state.fees.pix
        </p>
        <p>
          <strong>
            R$ 3,80 
          </strong>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`render pix payment fixed fee and spread fees 1`] = `
<div>
  <div>
    <p
      class="title"
    >
      pages.empty_state.fees.paid
    </p>
    <div
      class="feeValuesWrapper"
    >
      <div>
        <p>
          pages.empty_state.fees.pix
        </p>
        <p>
          <strong>
            R$ 3,80 + 1,19% 
          </strong>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`render pix payment spread fee 1`] = `
<div>
  <div>
    <p
      class="title"
    >
      pages.empty_state.fees.paid
    </p>
    <div
      class="feeValuesWrapper"
    >
      <div>
        <p>
          pages.empty_state.fees.pix
        </p>
        <p>
          <strong>
            1,19% 
          </strong>
        </p>
      </div>
    </div>
  </div>
</div>
`;
